### PR TITLE
feat(build): add some more meaningful error messages

### PR DESCRIPTION
**Pull request by max397574**

### CODE REVIEW

File: rocks-lib/src/rockspec/build/utils/mod.rs [hunk 1]
```diff
@@ -29,8 +29,18 @@ pub fn compile_c_files(files: &Vec<PathBuf>, target_file: &str, target_dir: &Pat
     let target = lua_module_to_pathbuf(target_file, std::env::consts::DLL_SUFFIX);
     let target = target_dir.join(target);
 
-    let parent = target.parent().expect("TODO");
-    let file = target.file_name().expect("TODO");
+    let parent = target.parent().unwrap_or_else(|| {
+        panic!(
+            "Couldn't determine parent for path {}",
```
Comment (max397574) [inline rocks-lib/src/rockspec/build/utils/mod.rs:34]: didn't see anything like it in the code
but do you perhaps have any standards for error messages? E.g. adding a prefix to indicate where they come from?

File: rocks-lib/src/rockspec/build/utils/mod.rs [hunk 2]
```diff
@@ -29,8 +29,18 @@ pub fn compile_c_files(files: &Vec<PathBuf>, target_file: &str, target_dir: &Pat
     let target = lua_module_to_pathbuf(target_file, std::env::consts::DLL_SUFFIX);
     let target = target_dir.join(target);
 
-    let parent = target.parent().expect("TODO");
-    let file = target.file_name().expect("TODO");
+    let parent = target.parent().unwrap_or_else(|| {
```
Comment (max397574) [inline rocks-lib/src/rockspec/build/utils/mod.rs:32]: https://rust-lang.github.io/rust-clippy/master/index.html#/expect_fun_call

File: rocks-lib/src/rockspec/build/utils/mod.rs (hunk 1 above)
Comment (vhyrro) [reply rocks-lib/src/rockspec/build/utils/mod.rs:34]: We'll be estabilishing a standard from error messages once me move over to the ariadne crate for everything. For now, this is fine :)

**Review by vhyrro — APPROVED** [review]

**Comment by max397574** [comment]
should I also write tests for this?

File: rocks-lib/src/rockspec/build/utils/mod.rs [hunk 3]
```diff
@@ -25,12 +25,24 @@ pub fn copy_lua_to_module_path(
 }
 
 /// Compiles a set of C files into a single dynamic library and places them under `{target_dir}/{target_file}`.
+/// # Panics
+/// Panics if no parent or no filename can be determined for the taraget path.
```
Comment (vhyrro) [inline rocks-lib/src/rockspec/build/utils/mod.rs:29]: typo: `taraget` :p

**Pull request merged**